library/src/webapp/js/lang-datepicker/lang-datepicker.js

```diff
@@ -1,2004 +1,377 @@
-        initVal = options.val;
+	// Default options
+const defaults = {
+	input: null,
+	useTime: true,
+	parseFormat: null,
+	allowEmptyDate: false,
+	val: "",
+	ashidden: {},
+	duration: null,
+	onDateTimeSelected: null,
+  };
+  
+  // DateHelper utility to standardize date handling
+  class DateHelper {
+    // Pad a number with leading zeros
+    static pad(num) {
+      return String(num).padStart(2, "0");
+    }
+    
+    // Convert any date representation to a standardized object
+    // IMPORTANT: We preserve local time exactly as entered, no timezone manipulation
+    static normalize(value) {
+      if (!value) return null;
+      
+      // Already a Date object - preserve its local time components
+      if (value instanceof Date) {
+        const d = new Date();
+        d.setFullYear(value.getFullYear());
+        d.setMonth(value.getMonth());
+        d.setDate(value.getDate());
+        d.setHours(value.getHours());
+        d.setMinutes(value.getMinutes());
+        d.setSeconds(value.getSeconds());
+        d.setMilliseconds(0);
+        return d;
       }
-
-      // if getval is set, this will override val
-      if (typeof options.getval !== 'undefined') {
-        initVal = $(options.getval).val();
+      
+      // Our custom date object format - use components directly
+      if (typeof value === 'object' && 'year' in value) {
+        const d = new Date();
+        d.setFullYear(value.year);
+        d.setMonth(value.month - 1); // JS months are 0-indexed
+        d.setDate(value.day);
+        d.setHours(value.hours || 0);
+        d.setMinutes(value.minutes || 0);
+        d.setSeconds(value.seconds || 0);
+        d.setMilliseconds(0);
+        return d;
       }
-
-      // finally trim the initVal and make sure it is set so we don't Dec 1969 the user
-      initVal = jQuery.trim(initVal);
-
-      if (!(initVal == "" && options.allowEmptyDate)){
-        if (!initVal) {
-          initVal = $.datepicker._getPreferredSakaiDatetime();
+      
+      // String parsing - try built-in parsing first
+      if (typeof value === 'string') {
+        // Handle YYYY-MM-DD HH:mm[:ss] format
+        if (value.includes(' ')) {
+          const [datePart, timePart] = value.split(' ');
+          const [year, month, day] = datePart.split('-').map(Number);
+          const [hours, minutes] = timePart.split(':').map(Number);
+          
+          if (!isNaN(year) && !isNaN(month) && !isNaN(day) && !isNaN(hours) && !isNaN(minutes)) {
+            const d = new Date();
+            d.setFullYear(year);
+            d.setMonth(month - 1);
+            d.setDate(day);
+            d.setHours(hours);
+            d.setMinutes(minutes);
+            d.setSeconds(0);
+            d.setMilliseconds(0);
+            return d;
+          }
+        }
+        
+        // Handle ISO format (YYYY-MM-DDTHH:mm)
+        if (value.includes('T')) {
+          const [datePart, timePart] = value.split('T');
```

Comment:
Could you not use the polyfilled Temporal stuff here?

Temporal.PlainDateTime.from(value);

https://developer.mozilla.org/en-US/blog/javascript-temporal-is-coming/

Reply:
happy to reduce code once Temporal is widely available in browsers. But loading a big Polyfill dependency to reduce a few lines doesn't make sense to me.

But yeah Temporal looks very awesome and much needed.

Reply:
It's already being used in sakai-date-picker, so we're loading it already. Well, it's being bundled.

Reply:
can i depend on it being there? if that's a safe assumption, i'm happy to use it

Reply:
It's brought in as an es module and thus bundled. However, the methods would need to be referenced on globalThis to be usable from non module code. We do that for other stuff in the bundle config in webcomponents, so we could. Latest FF has implemented it but we're bundling so feature detection won't yield any benefit.

Reply:
It takes me ages to type this with my left hand so I hope you appreciate it ;)

Reply:
For what it's worth I think you should migrate these hard won smarts into sakai-date-picker and just emit that tag from the date picker function. Unless I'm missing something obvious it seems that that would work. That way we move the entire codebase to the native picker and we can migrate at our leisure later, if we wish.

Reply:
I agree.... that would be a nice victory and it's where the code should live.

I don't think I'm up for spending another 20-30 hours on this right now. I'm willing to do it down the road. 

I think ultimately this is an improvement that's worth merging. It gets us closer to where we need to be. It reduces our dependence on jquery-ui and outside libs. It moves us closer to where we should be.

Reply:
Agreed :)

Reply:
and thanks for bringing up Temporal again. it's an awesome API ..... solves all of my complaints about Dates in JS ... *all*!